bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -480,8 +480,12 @@ public long addEntry(ByteBuf entry) throws IOException, BookieException {
         long stamp = writeCacheRotationLock.tryOptimisticRead();
         boolean inserted = false;
 
-        inserted = writeCache.put(ledgerId, entryId, entry);
-        if (!writeCacheRotationLock.validate(stamp)) {
+        // If the stamp is 0, the lock was exclusively acquired, validation will fail, and we can skip this put.
```

Comment:
I looked closer, and it's only 0 when the lock is exclusively held, which only happens when we call `swapWriteCache()`. That does not seem frequent enough to bother with this special case, but I will wait for others to review to confirm.

Reply:
It seems like there's invalid use of the StampedLock before this PR. Read operations are used for write operations. That's not supported by StampedLock. I don't see how that could be useful.

Reply:
I think it's better to just use the StampedLock API to do that validation. It's not an expensive operation.